test/unit/create-federation-secret-job.bats

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env bats
+
+load _helpers
+
+@test "createFederationSecet/Job: disabled by default" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/create-federation-secret-job.yaml  \
+      . | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
+  [ "${actual}" = "false" ]
+}
+
+@test "createFederationSecet/Job: fails when global.federation.enabled=false" {
+  cd `chart_dir`
+  run helm template \
+      -x templates/create-federation-secret-job.yaml  \
+      --set 'global.federation.createFederationSecret=true' .
+  [ "$status" -eq 1 ]
+  [[ "$output" =~ "global.federation.enabled must be true when global.federation.createFederationSecret is true" ]]
+}
+
+# NOTE: This error actually comes from server-statefulset but we test it here
```

Comment:
Not a huge deal, but I think it's more intuitive if we test it in server tests because it should fail even when you set `createFederationSecret` to `false` and the failure code is in the server stateful set too.

Reply:
I was thinking it made sense to be in both? If it's not in the server as well I'll add it. I was thinking that because this spec requires that fail to be in place, if it's only tested in the server then if we disable it there we won't know that this job also requires it.

Reply:
Oh! yep it is tested in both places; I misenterpreted the comment. disregard, please 😄

Reply:
I've added the missing test to server but kept this one here too. lmk.